exchanges/stream/websocket_test.go

```diff
@@ -355,7 +503,14 @@ func TestConnectionMonitorNoConnection(t *testing.T) {
 	ws.ShutdownC = make(chan struct{}, 1)
 	ws.exchangeName = "hello"
 	ws.trafficTimeout = 1
-	go ws.connectionMonitor()
+	ws.verbose = true
+	ws.Wg = &sync.WaitGroup{}
+	ws.connectionMonitor()
+	ws.connectionMonitor() // This one should exit
+	time.Sleep(time.Second)
+	ws.connected = true  // attempt shutdown when not enabled
```

Comment:
Any reason here not to use the lock functions? If they ever get flipped while this test is running you could get a data race